tests/Unit/Helpers/ControlSystem/Examples.hpp

```diff
@@ -120,7 +121,8 @@ struct ExampleControlError
 struct ExampleControlSystem
     : tt::ConformsTo<control_system::protocols::ControlSystem> {
   static std::string name() { return "ExampleControlSystem"; }
-  static std::string component_name(const size_t i) {
+  static std::optional<std::string> component_name(
+      const size_t i, const size_t /*num_components*/) {
```

Comment:
Should the example perhaps do something with num_components?  Like check that it is <=3?